src/control/server/mgmt_svc.go

```diff
@@ -212,9 +212,9 @@ func (svc *mgmtSvc) Join(ctx context.Context, req *mgmtpb.JoinReq) (*mgmtpb.Join
 		svc.members = append(svc.members, &SystemMember{
 			Addr: req.GetAddr(), Uuid: req.GetUuid(), Rank: resp.GetRank(),
 		})
-		svc.mutex.Unlock()
 
 		svc.log.Debugf("MgmtSvc.members: %+v\n", svc.members)
+		svc.mutex.Unlock()
```

Comment:
this was requested by @mjmac , I will make it consistent

Reply:
To be clear, the lock isn't to protect the logger, it's to protect svc.members from unsynchronized reads. Rather than sprinkling locks around everywhere, though, I'd recommend creating a new type to wrap the membership slice (or map, so that you can approximate a set) with synchronized access methods.

Reply:
ok, will do that